src/org/sosy_lab/common/NativeLibraries.java

```diff
@@ -105,55 +110,53 @@ public static Architecture guessVmArchitecture() {
         return currentArch;
       }
 
-      @Var String prop = System.getProperty("os.arch.data.model");
-      if (isNullOrEmpty(prop)) {
-        prop = System.getProperty("sun.arch.data.model");
+      currentArch = downgradeArchIfRequired(getArchFromSystemProperty());
```

Comment:
The `downgradeArchIfRequired()` here is misleading. We are not "downgrading" an architecture that we already have.

What we do here is that we guess the JVM architecture. And one of the inputs for this heuristic happens to be the OS architecture. So the logic for guessing the JVM architecture should be in a method whose name reflects that, and the OS architecture should be in a variable/parameter named `osArch`.

My suggestion is to move the logic for guessing the JVM architecture into `guessVmArchitecture()` and do `Architecture osArch = guessOsArchitecture();`. If you want to keep more parts of the logic in separate methods than everything in one method as it was (seems like a good idea), we could have a method `isVm32Bit()`. That should make the logic in `guessVmArchitecture()` pretty easy to follow.